docs/DeveloperGuide.adoc

```diff
@@ -235,12 +315,91 @@ image::CommitActivityDiagram.png[]
 
-_{Explain here how the data encryption feature will be implemented}_
+==== Current Implementation
+Given below is the execution process of the `NotificationCommand`:
 
-// end::dataencryption[]
+1. The user executes `add -b /name John ...` to add a new body to Mortago.
+2. The add command also created a new new `Notif` using this body as the parameter.
+3. The add command instantiates a new`NotifCommand` with `Notif`, `period`, `unit` as parameters.
+`period` is a static final integer initialized for 10 seconds and `unit` is static final TimeUnit set at TimeUnit.SECONDS.
+4. `NotifCommand` creates a new static `ScheduledExecutorService` which is scheduled to update the `BodyStatus` to
+`CONTACT_POLICE` if there is no change in its status for the specified `period` using the `UpdateCommand`.
+5. After the `BodyStatus` is updated, a `NotifWindow` will be shown to alert the user of the automatic update in
+`BodyStatus`.
+
+.Notification Command Sequence Diagram
+image::NotifCommandSequenceDiagram.png[]
+
+.ChangeUI function Sequence Diagram
+image::ChangeUiFnSequenceDiagram.png[]
+
+==== Design Considerations
+===== Aspect: How to delay change in BodyStatus
+* **Alternative 1 (current choice):** Use `ScheduledExecutorService`.
+** Pros: Does not depend on thread synchronization and avoids the need to deal with threads directly.
+** Cons: May cause memory leaks if cache is not cleared.
+
+* **Alternative 2:** Use `Thread.sleep`
+** Pros: Straightforward way to delay a thread.
+** Cons: May quickly run into OutOfMemory error.
+
+==== Aspect: How to reflect change in BodyStatus on UI
+* **Alternative 1 (current choice):** Use `UpdateCommand`.
+** Pros: Easy to implement.
+** Cons: Increases `NotifCommand`'s dependency.
+
+* **Alternative 2: Directly make changes to the model.
+** Pros: Does not increase dependency .
+** Cons: Cause a lot of repetition of code.
+
+// tag::genpdf[]
+=== Generate PDF feature
+==== Implementation
+
+The generate PDF feature is facilitated by `ReportGenerator`.
+It extends `Mortago` with the ability to create a report, supported by iText external library.
+Additionally, it implements the following operations:
+
+* `ReportGenerator#generate(body)` -- Creates report in a PDF file for the specific body.
+* `ReportGenerator#generateAll()` -- Creates report in a PDF file for all bodies registered in Mortago.
+
+The following sequence diagram shows how the execute operation works:
+
+image::GenReportSequenceDiagram.png[]
+
+NOTE: The lifeline for `GenReportCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+The `genReport <BODY_ID>` command calls `ReportGenerator#execute()`, which creates the document.
+
+The following activity diagram summarizes what happens when a user executes a `genReport <BODY_ID>` command:
+
+image::GenReportActivityDiagram.png[]
```

Comment:
Perhaps this could be more specific? 

Looks too similar with the above Activity Diagram. More differentiation might be needed.